blockchain/types/account/account_serializer.go

```diff
@@ -127,3 +141,29 @@ func (ser *AccountSerializer) UnmarshalJSON(b []byte) error {
 
 	return json.Unmarshal(dec.Account, ser.account)
 }
+
+// Unextend ExtHash fields within an RLP-encoded account.
+// If the supplied bytes is not an RLP-encoded account, or does not contain any ExtHash,
+// then return the supplied bytes unchanged.
+func UnextendRLP(b []byte) []byte {
+	if len(b) <= common.ExtHashLength {
```

Comment:
I guess both are fine because accounts are RLP encoded, where at least one more byte is prepended before a hash.
But your suggestion is more intuitive.

Reply:
Understand. It would be helpful to provide additional comments.

Upon reviewing the question again, I believe that the condition should be compared with the size of the `SmartContractAccount`. If the size of any RLP-encoded SCAs exceeds the `common.ExtHashLength`, it would result in an early exit with unchanged bytes. Is it correct?

Reply:
The original reasoning was that if it's shorter than ExtHash, then there is nothing to unextend.
But you are right. The threshold can be the shortest SCA length instead. Then we can early exit some long EOA (e.g. containing AccountKey).

It turns out the shortest possible encoded SCA is 43 bytes long. 43 bytes is not much different from ExtHashLength (39). So I suggest just changing it to `len(b) < ExtHashLength` and keep the code intuitive.

```go
        commonFields := &AccountCommon{
                nonce:         0,
                balance:       big.NewInt(0),
                humanReadable: false,
                key:           accountkey.NewAccountKeyLegacy(),
        }
        sca := &SmartContractAccount{
                AccountCommon: commonFields,
                storageRoot:   common.Hash{}.ExtendLegacy(),
                codeHash:      nil,
                codeInfo:      params.CodeInfo(0),
        }

        enc := NewAccountSerializerWithAccount(sca)
        b, _ := rlp.EncodeToBytes(enc)
        fmt.Printf("(%d) %x\n", len(b), b)
        // (43) 02e9c580808001c0a000000000000000000000000000000000000000000000000000000000000000008080
```

Reply:
Then, how about removing it? Removing this comparison would have more benefit for readability, compared to its performance contribution.

Reply:
We have a case where `valueNode` is a contract storage slot (33 byte RLP-encoded). This happens during a normal operation. Without the if statement, UnserializeSerializedAccount crashes.

I think I need a non-crashing heuristic test to check if the given byte blob is a serialized account or not.

Reply:
I removed the length conditional. Instead, I made UnextendSerializedAccount to never panic. PTAL ca9f49d.

Reply:
Ah, thanks. Looks more clear.